src/pages/Home/components/Portfolio/Prompt/Prompt.tsx

```diff
@@ -0,0 +1,508 @@
+        ) {
+          throw new Error(`Cannot find the destination token`);
+        }
+
+        const amountBigInt = stringToBigint(
+          amount.toString(),
+          srcToken.decimals,
+        );
+
+        const rate = await getRate({
+          srcDecimals: srcToken.decimals,
+          srcToken:
+            srcToken.type === TokenType.ERC20
+              ? srcToken.address
+              : srcToken.symbol,
+          destToken:
+            toToken.type === TokenType.ERC20 ? toToken.address : toToken.symbol,
+          destDecimals: toToken.decimals,
+          srcAmount: amountBigInt.toString(),
+        });
+        if (
+          isAPIError(rate.optimalRate) ||
+          !rate.optimalRate ||
+          !rate.destAmount
+        ) {
+          throw new Error(
+            `Error while getting swap rate. ${isAPIError(rate.optimalRate) ? rate.optimalRate.message : ''}`,
+          );
+        }
+
+        await swap({
+          srcToken:
+            srcToken.type === TokenType.ERC20
+              ? srcToken.address
+              : srcToken.symbol,
+          destToken:
+            toToken.type === TokenType.ERC20 ? toToken.address : toToken.symbol,
+          srcDecimals: srcToken.decimals,
+          destDecimals: toToken.decimals,
+          srcAmount: rate.optimalRate.srcAmount,
+          priceRoute: rate.optimalRate,
+          destAmount: rate.destAmount,
+          slippage: 0.15,
+        });
+
+        return {
+          content: `Swap initiated ${amount}${srcToken.symbol} to ${rate.destAmount}${toToken.symbol}.`,
+        };
+      },
+    }),
+    [
+      accounts.active,
+      createContact,
+      getRate,
+      network,
+      request,
+      selectAccount,
+      swap,
+      t,
+      tokens,
+    ],
+  );
+
+  const systemPrompt = useMemo(() => {
+    if (!network || !tokens) {
+      return '';
+    }
+    return systemPromptTemplate
+      .replace(
+        '__TOKENS__',
+        JSON.stringify(
+          tokens.map((token) => ({
+            name: token.name,
+            symbol: token.symbol,
+            balance: token.balanceDisplayValue,
+          })),
+          (_, v) => (typeof v === 'bigint' ? v.toString() : v),
+        ),
+      )
+      .replace(
+        '__NETWORKS___',
+        JSON.stringify(
+          networks.map((n) => ({
+            id: n.caipId,
+            name: n.chainName,
+            isTestnet: n.isTestnet,
+          })),
+        ),
+      )
+      .replace(
+        '__CURRENT_NETWORK___',
+        JSON.stringify({
+          id: network.caipId,
+          name: network.chainName,
+          isTestnet: network.isTestnet,
+        }),
+      )
+      .replace('__CONTACTS__', JSON.stringify(contacts))
+      .replace(
+        '__ACCOUNTS__',
+        JSON.stringify(
+          [
+            ...Object.values(accounts.primary).flat(),
+            ...Object.values(accounts.imported),
+          ].map((a) => ({ name: a.name, id: a.id, address: a.addressC })),
```

Comment:
We're telling the model:

> The active account is marked with the active property:

But the `active` property is not provided here.

```suggestion
          ].map((a) => ({
            name: a.name,
            id: a.id,
            address: a.addressC,
            active: a.id === accounts.active?.id,
          })),
```